src/foundry/common/primitives/math.mjs.d.ts

```diff
@@ -61,4 +61,17 @@ interface Math {
    * @returns An angle in radians
    */
   toRadians(angle: number): number;
+
+  /**
+   * Get an oscillation between lVal and hVal according to t
+   * @param  minVal - The minimal value of the oscillation.
+   * @param  maxVal - The maximum value of the oscillation.
+   * @param  t      - The time value.
+   * @param  p      - The period (can't be equal to 0).
+   *                  (default: `1`)
+   * @param  func   - The optional math function to use for oscillation.
+   *                  (default: `Math.cos`)
+   * @returns The oscillation according to t.
+   */
+  oscillation(minVal: number, maxVal: number, t: number, p: number, func: (radians: number) => number): number;
```

Comment:
Because `p` and `func` have defaults they're both optional parameters. The way of notating that in Typescript is `p?: number` and `func?: (radians: number) => number` where the `?

Reply:
Yeah that was just a forgetful mistake, I've learned that bit from working on SWADE